src/bases.jl

```diff
@@ -489,62 +476,113 @@ requires either a dual basis or the cached basis to be selfdual, for example ort
 See also: [`get_vector`](@ref), [`get_basis`](@ref)
 """
 function get_coordinates(M::AbstractManifold, p, X, B::AbstractBasis)
-    Y = allocate_result(M, get_coordinates, p, X, B)
-    return get_coordinates!(M, Y, p, X, B)
+    return _get_coordinates(M, p, X, B)
+end
+
+function _get_coordinates(M::AbstractManifold, p, X, B::VeeOrthogonalBasis)
+    return get_coordinates_vee(M, p, X, number_system(B))
+end
+function get_coordinates_vee(M::AbstractManifold, p, X, N)
+    return get_coordinates(M, p, X, DefaultOrthogonalBasis(N))
 end
-@decorator_transparent_signature get_coordinates(
-    M::AbstractDecoratorManifold,
+
+function _get_coordinates(M::AbstractManifold, p, X, B::DefaultBasis)
+    return get_coordinates_default(M, p, X, number_system(B))
+end
+function get_coordinates_default(M::AbstractManifold, p, X, N)
```

Comment:
```suggestion
function get_coordinates_default(M::AbstractManifold, p, X, N::AbstractNumbers)
```
What do you think about adding this type annotation here? This seems correct to me.

Reply:
Sure, sounds good.